### PR TITLE
[common] Switch to mbedtls v3.6.1

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

**Review by mlohvynenko — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

File: src/common/crypto/mbedtls/drivers/psa_crypto.c [hunk 1]
```diff
@@ -1,9176 +0,0 @@
-/*
```
Comment (al1img) [inline src/common/crypto/mbedtls/drivers/psa_crypto.c:1]: Why this files is removed? It should be updated instead.

File: suppressions.txt [hunk 2]
```diff
@@ -1,4 +1,3 @@
 useStlAlgorithm
 unusedFunction
-toomanyconfigs
```
Comment (al1img) [inline suppressions.txt:3]: Why?

File: src/common/crypto/mbedtls/drivers/psa_crypto.c (hunk 1 above)
Comment (mykola-kobets-epam) [reply src/common/crypto/mbedtls/drivers/psa_crypto.c]: done

File: suppressions.txt (hunk 2 above)
Comment (mykola-kobets-epam) [reply suppressions.txt:3]: produced cpp-check error. might be due to missing psa_driver.c
https://github.com/aosedge/aos_core_lib_cpp/actions/runs/10739968945/job/29786976717
replaced by suppressing psa-driver dir entirely as it produces other warnings:

**Pull request merged**